launcher/container_runner.go

```diff
@@ -236,14 +266,38 @@ func NewRunner(ctx context.Context, cdClient *containerd.Client, token oauth2.To
 		return nil, fmt.Errorf("failed to create REST verifier client: %v", err)
 	}
 
+	// Fetch container image signatures by using signaturediscovery client.
+	signaturesFetcher := func(ctx context.Context) []internal.Signature {
+		authToken, err := RetrieveAuthToken(mdsClient)
+		if err != nil {
+			logger.Printf("failed to retrieve auth token: %v, using empty auth for image pulling\n", err)
+		}
+		updatedSpec, err := spec.GetLaunchSpec(mdsClient)
```

Comment:
Why do you need to update the spec? The launcher should only ever get the spec at startup. Refetching it can lead to weird states and potential security issues.

Reply:
The rational behind updating the spec is to enable operators don't have to restart VM when new docker repos added. Even though ADMC team is ok with restarting VM when new repos added, I feel it might be nice to have this feature. But if refetching it will lead to any potential security issues, we should definitely avoid that.